src/lib/components/LighthouseScoresContainer/index.js

```diff
@@ -6,39 +8,79 @@
 class LighthouseScoresContainer extends HTMLElement {
   constructor() {
     super();
-    this.firstUpdated = false;
+
+    this.metaElement = null;
+    this.statsElement = null;
+    this.metricsElement = null;
+    this.auditsElement = null;
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+
+    this.onCategoryChanged = (ev) => {
+      this.auditsElement.filteringOn = ev.detail;
+    };
   }
 
-  connectedCallback() {
-    // LighthouseScoresContainer expects to find children elements which it manages. It's not a
-    // LitElement, so wire things up once it's connected, and keep track of whether it's done.
-    if (this.firstUpdated) {
+  onStateChanged() {
+    const {
+      lighthouseError,
+      lighthouseResult,
+      activeLighthouseUrl,
+    } = store.getState();
+
+    if (!lighthouseResult) {
+      // TODO: clear data?
```

Comment:
is this asking if we should clear out the user's urls in the store or in the input field? Seems like we'd want to leave everything in place and just show the error message so they can tweak what they've already typed in. But I'm probably misunderstanding :)

Reply:
Your "Seems like..." is what happens now. I'll update that so it's clear.